### PR TITLE
fix: make prover-service demo work again

### PR DESCRIPTION
**Pull request by mrain**
closes: #1161 

all clear

~~Stucked: `gen-demo-genesis` needs `SchnorrPrivKey` but it's in sequencer crate which hotshot-state-prover crate depends on.~~
~~Solution: remove `SchnorrPrivKey`, just use hotshot/jf types~~

### CODE REVIEW

**Comment by sveitser** [comment]
@mrain Are you blocked on this PR?

**Comment by mrain** [comment]
> @mrain Are you blocked on this PR?

Yes. Just pushed my commits. There're some new HotShot changes about `BlockPayload` that I don't know how to resolve

File: hotshot-state-prover/src/service.rs [hunk 1]
```diff
@@ -58,40 +62,83 @@ pub struct StateProverConfig {
...
+                tracing::info!("Peers' keys are not ready.");
+            }
+            Err(e) => {
+                num_retries -= 1;
+                tracing::warn!("Orchestrator error {e}, retrying.");
+            }
+        }
+        sleep(Duration::from_secs(2)).await;
+    }
+    panic!("Failed initializing stake table.")
```
Comment (nomaxg) [inline hotshot-state-prover/src/service.rs:141]: I think that this can retry indefinitely, don't see why we need a retry limit here

**Review by nomaxg — APPROVED** [review]
LGTM, just one comment

File: hotshot-state-prover/src/service.rs (hunk 1 above)
Comment (mrain) [reply hotshot-state-prover/src/service.rs]: changed in [43e59c7](https://github.com/EspressoSystems/espresso-sequencer/pull/1183/commits/43e59c71050e5be3330952d4b273289699775292)

**Pull request merged**